src/engine/ScreenShotTaker.cs

```diff
@@ -8,12 +8,23 @@
 public class ScreenShotTaker : NodeWithInput
 {
     private static ScreenShotTaker instance;
+    private bool isCurrentlyTakingScreenshot;
+    private Image screenshotImage;
+    private Steps step;
 
     private ScreenShotTaker()
     {
         instance = this;
     }
 
+    private enum Steps
```

Comment:
```suggestion
    private enum Step
```
I'd name this like that so that `Step.Start` reads more naturally to me.